### ISSUE TITLE
bug in tab group, default group assignment is incorrect.

### ISSUE THREAD

**Issue by serweb-labs** [bolt/bolt#2359]
The incidence occurs when some of the fields do not have a defined group, rather than grouped into "ungrouped" appear in the last group of the iteration.

for example.
```yml
orders:
    name: orders
    singular_name: order
    fields:
        title:
            type: text
            class: large
        image:
            type: image
            group: media
        message:
            type: html
            height: 300px
```
In this case the first go to "Ungrouped", the second "Media" and also the third show in "Media" and the latter is the error, apparently the attribute data-tab is assigned to the last variable instead of "ungrouped" .

merry christmas (jo jo jo), and I wish that consumerism does not take over this party ^^

**Comment by rarila** [comment]
@serweb-labs To be honest, this isn't a bug, it's a feature.
It's designed that way, that all fields that follows after a field with a defined group inherit it's group. That way you have to enter every group only once. You can love or hate it…

**Comment by rarila** [comment]
@bobdenotter So that we found out that it is no bug, how would we handle this? Declare it as not a  bug? Declare it as bug, changing behaviour with BC break? :coffee: or :beer:

**Comment by serweb-labs** [comment]
@rarila Thanks for the reply, as what  @rossriley  says, what can you say?

**Comment by rarila** [comment]
@serweb-labs a duplicate of #2424… working on it

**Comment by serweb-labs** [comment]
Thanks for the clarification.

**Comment by rarila** [comment]
@serweb-labs should the behavior be changed or is it more a matter of documentation?

**Comment by serweb-labs** [comment]
Actually I think a matter of behavior, but it is only an opinion and I am very respectful of team decisions.

**Comment by rarila** [comment]
I think it was more kind of a let's do it than a team decision ;-)

**Comment by rarila** [comment]
Ok, just saw you added those tabs in #1115 and @bobdenotter added this behavior in 0ecc2ab864c450d491689aed024fb4919e1fde0c

**Comment by serweb-labs** [comment]
I think it's great to have an automatic behavior, and I think it can be polished further. :)

**Comment by rarila** [comment]
So what could be polished?

**Comment by rarila** [comment]
[RFC] Another idea: As the ``Ungrouped`` group isn't there by design but by necessity to catch ungrouped fields until the first grouped fields and is only meant to be seen in the development process anyway, why not simply dump it and attach those orphans simply at the first defined group.

That would save us a not really required translation and would probably avoid confusion why fields with no group don't appear in the unprouped tab as there never is one.

So the following …

```yml
    fields:
        field_m1:
            type: text
        field_m2:
            type: image
            group: media
        field_m3:
            type: text
        field_f1:
            type: html
            group: foo
        field_f2:
            type: html
        field_b1:
            type: html
            group: bar
        field_m4:
            type: image
            group: media
```
…  would result give us this structure:

```
    Media:
        field_m1
        field_m2
        field_m3
        field_m4
    Foo:
        field_f1
        field_f2
    Bar:
        field_b1
```

**Comment by GawainLynch** [comment]
To be honest, I would prefer we had gone, or go with, something like this:

```yaml
pages:
    name: Pages
    slug: pages
    singular_name: Page
    singular_slug: page
    fields:
        field_m1:
            type: text
        field_m2:
            type: image
        field_m3:
            type: text
        field_f1:
            type: html
        field_f2:
            type: html
        field_b1:
            type: html
        field_m4:
            type: image
    field_groups:
        media: [ field_m2, field_m4 ]
        foo: [ field_f1 ]
        bar: [ field_b1 ] 
```

With the first array element key of `field_groups` being the *default* landing place for anything not in one of those array elements.

Separation of concerns/logic etc.

**Comment by rarila** [comment]
@GawainLynch problem is the word _had_. BC hits :facepunch:

And your proposal would mean similar extra discussion on what should happen with ``fields`` not in ``field_groups``, ``fields`` in ``field_groups`` not in ``fields`` …

So we had to refactor and deprecate anyway. My proposal was to get into 2.1 and disarm the situation a bit without (major) breaking stuff.

**Comment by GawainLynch** [comment]
> So we had to refactor and deprecate anyway. My proposal was to get into 2.1 and disarm the situation a bit without (major) breaking stuff.

Personally, I wouldn't be against a change like that, if it is not too far away and the code changes weren't large.

**Comment by bobdenotter** [comment]
I'm biased, because I made it how it is now, but I like it, the way it is now. :-)

**Comment by rarila** [comment]
@bobdenotter Nobody told you to read THAT book! ;-)

Ok, then you should at least remove the not fitting labels :)

**Comment by serweb-labs** [comment]
I like the point of view of @GawainLynch . :1:

**Comment by rarila** [comment]
@serweb-labs didn't you invent the format we have now ;-)

Anyway, that can't be changed for 2.1. RFCs for a format to be implemented in the next BC breaking version are welcome, I guess :beers:

**Comment by serweb-labs** [comment]
:+1:  Anyway the improvements made are bienvienidas and leave me satisfied as a user.

**Comment by rarila** [comment]
Ok, then I'm satisfied as a developer. Can we close then?

**Comment by serweb-labs** [comment]
Thanks!

**Comment by rarila** [comment]
No problem!
